librad/src/git/replication.rs

```diff
@@ -539,22 +543,45 @@ mod person {
         storage: &Storage,
         urn: &Urn,
         delegates: BTreeSet<PeerId>,
-    ) -> Result<(), Error> {
-        let persons = NonEmpty::from_vec(
-            delegates
-                .into_iter()
-                .flat_map(|peer| {
-                    let urn =
-                        unsafe_into_urn(Reference::rad_id(Namespace::from(urn)).with_remote(peer));
-                    identities::person::get(storage, &urn).ok().flatten()
-                })
-                .collect(),
-        );
-        let tip = match persons {
-            Some(persons) => identities::person::latest_tip(storage, persons).map_err(Error::from),
-            None => Err(Error::MissingIdentities(urn.clone())),
-        }?;
-        ensure_rad_id(storage, urn, tip.into())
+    ) -> Result<IdStatus, Error> {
+        use IdStatus::*;
+
+        let local_peer = storage.peer_id();
+        let delegates: BTreeMap<PeerId, VerifiedPerson> = delegates
+            .into_iter()
+            .map(|peer| {
+                let remote_urn =
+                    unsafe_into_urn(Reference::rad_id(Namespace::from(urn)).with_remote(peer));
+                let verified = identities::person::verify(storage, &remote_urn)?
+                    .ok_or_else(|| Error::MissingIdentities(remote_urn.clone()))?;
+
+                Ok((peer, verified))
+            })
+            .collect::<Result<_, Error>>()?;
+        let latest = {
+            let mut prev = None;
+            for pers in delegates.values().cloned() {
+                match prev {
+                    None => prev = Some(pers),
+                    Some(p) => {
+                        let newer = identities::person::newer(storage, p, pers)?;
+                        prev = Some(newer);
+                    },
+                }
+            }
+            prev.expect("empty delegations")
```

Comment:
Hrm... I was convinced that we couldn't get here with empty delegates, but maybe we can? There's a `NoTrustee` error variant, but it doesn't seem to get constructed ever (and `rustc` doesn't warn **/me shakes fist**)

Reply:
Ya, I just scanned through `identities::git` and I don't see any check on the delegations :grimacing:

Reply:
It cannot happen(tm), because we run `verify` before getting here. Just to be sure, I added a `cmp::max(1, ..)` to `quorum_threshold()`. I think I'll cook up a simple `NonEmptySet` next.

Reply:
:man_cook: